src/App.jsx

```diff
@@ -14,45 +41,67 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isAscending, setIsAscending] = useState(true);
+
+  const toggleSortOrder = () => {
+    setIsAscending(!isAscending);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sortedGoods = getPreparedGoods(goodsFromServer, sortField);
+  const visibleGoods = isAscending ? sortedGoods : sortedGoods.reverse();
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': isAscending,
+          })}
+          onClick={toggleSortOrder}
+        >
+          Reverse
+        </button>
+
+        {
+          (sortField !== '' || !isAscending) && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
```

Comment:
Consider moving this logic from jsx to a separate function
```suggestion
              onClick={reset}
```